orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CancelExecutionHandler.kt

```diff
@@ -51,8 +46,6 @@ class CancelExecutionHandler(
 
       // then, make sure those runTask messages get run right away
       queue.push(RescheduleExecution(execution))
-
-      publisher.publishEvent(ExecutionComplete(this, message.executionType, message.executionId, CANCELED))
```

Comment:
this was happening too early when published explicitly here - it ends up being fired from the CompleteExecutionHandler now